File: vibe_coding/nxn_table.cpp
#include <iostream>
#include <iomanip>
#include <string>

using namespace std;

void printNxNTable(int n) {
    // Calculate the width needed for formatting
    int maxValue = n * n;
    int width = to_string(maxValue).length() + 1;
    
    // Print header row
    cout << setw(width) << " ";
    for (int i = 1; i <= n; i++) {
        cout << setw(width) << i;
    }
    cout << endl;
    
    // Print separator line
    cout << setw(width) << " ";
    for (int i = 1; i <= n; i++) {
        cout << string(width, '-');
    }
    cout << endl;
    
    // Print table rows
    for (int i = 1; i <= n; i++) {
        // Print row header
        cout << setw(width) << i << "|";
        
        // Print table values
        for (int j = 1; j <= n; j++) {
            cout << setw(width) << (i * j);
        }
        cout << endl;
    }
}

int main() {
    int n;
    
    cout << "Enter the size of the NxN table: ";
    cin >> n;
    
    if (n <= 0) {
        cout << "Error: N must be a positive integer." << endl;
        return 1;
    }
    
    cout << "\n" << n << "x" << n << " Multiplication Table:\n" << endl;
    printNxNTable(n);
    
    return 0;
}

File: vibe_coding/nxn_table.py
def print_nxn_table(n):
    """
    Print an NxN multiplication table with proper formatting.
    """
    # Calculate the width needed for formatting
    max_value = n * n
    width = len(str(max_value)) + 1
    
    # Print header row
    print(f"{'':>{width}}", end="")
    for i in range(1, n + 1):
        print(f"{i:>{width}}", end="")
    print()
    
    # Print separator line
    print(f"{'':>{width}}", end="")
    print("-" * (width * n))
    
    # Print table rows
    for i in range(1, n + 1):
        # Print row header
        print(f"{i:>{width}}|", end="")
        
        # Print table values
        for j in range(1, n + 1):
            print(f"{i * j:>{width}}", end="")
        print()


def main():
    try:
        n = int(input("Enter the size of the NxN table: "))
        
        if n <= 0:
            print("Error: N must be a positive integer.")
            return
        
        print(f"\n{n}x{n} Multiplication Table:\n")
        print_nxn_table(n)
        
    except ValueError:
        print("Error: Please enter a valid integer.")


if __name__ == "__main__":
    main()

